lib/modules.js

```diff
@@ -145,21 +151,33 @@ export async function updateSourceFiles ({
     }
   }
 
-  const url = `https://${cid}.ipfs.w3s.link?format=car`
-  console.error(`[${module}]  ⇣ downloading source files via ${url}`)
-  const res = await fetch(url)
-
-  if (res.status >= 300) {
-    throw new Error(
-      `[${module}] Cannot fetch ${module} archive for ${cid}: ${res.status}\n` +
-      await res.text()
-    )
-  }
+  let res
+  for (const gateway of gateways) {
+    try {
+      const url = `https://${cid}.ipfs.${gateway}?format=car`
+      console.error(`[${module}]  ⇣ downloading source files via ${url}`)
+      res = await fetch(url)
+
+      if (res.status >= 300) {
+        throw new Error(
+          `[${module}] Cannot fetch ${module} archive for ${cid}: ${res.status}\n` +
+          await res.text()
+        )
+      }
 
-  if (!res.body) {
-    throw new Error(
-      `[${module}] Cannot fetch ${module} archive for ${cid}: no response body`
-    )
+      if (!res.body) {
+        throw new Error(
+          `[${module}] Cannot fetch ${module} archive for ${cid}: no response body`
+        )
+      }
+      break
+    } catch (err) {
+      if (gateway === gateways[gateways.length - 1]) {
+        throw err
```

Comment:
I would throw a different error explaining that we have exhausted all fallback options to avoid potential confusion.

Reply:
👍  https://github.com/filecoin-station/core/pull/346